tests/templates/test_state_preparations/test_state_prep_mps.py

```diff
@@ -526,6 +527,146 @@ def test_correctness(self, mps, state, num_wires, num_work_wires):
+                                [-0.20758717, 0.1329479, -0.18184107, 0.06942658],
+                            ],
+                            [
+                                [-0.16499137, -0.13680142, -0.18432824, 0.12950892],
+                                [-0.68790868, -0.64141472, -0.12485688, -0.0556177],
+                            ],
+                            [
+                                [0.0352582, -0.37993402, 0.26781956, -0.25935129],
+                                [0.04351872, -0.27109361, 0.65111429, 0.4648453],
+                            ],
+                            [
+                                [0.1909576, 0.25461839, -0.07463641, -0.34390477],
+                                [-0.21279487, 0.0305474, 0.53420894, -0.66578494],
+                            ],
+                        ]
+                    ),
+                    np.array(
+                        [
+                            [[-0.26771292, -0.00628612], [-0.96316273, 0.02465422]],
+                            [[0.96011241, 0.07601506], [-0.2663889, 0.03798452]],
+                            [[-0.00727353, 0.4537835], [-0.02374101, -0.89076596]],
+                            [[0.08038064, -0.88784161], [-0.02812246, -0.45220057]],
+                        ]
+                    ),
+                    np.array([[-0.97855153, 0.2060022], [0.2060022, 0.97855153]]),
+                ],
+                5,
+            ),
+        ],
+    )
+    def test_decomposition_new(self, mps, num_wires):
+        """Tests the decomposition rule implemented with the new system."""
+        op = qml.MPSPrep(
+            mps, wires=range(2, num_wires + 2), work_wires=[0, 1], right_canonicalize=True
+        )
+        for rule in qml.list_decomps(qml.MPSPrep):
+            _test_decomposition_rule(op, rule)
+
+    @pytest.mark.capture
+    @pytest.mark.usefixtures("enable_graph_decomposition")
+    def test_decomposition_capture(self):
+        """Tests that the new decomposition works with capture."""
+
+        from pennylane.tape.plxpr_conversion import CollectOpsandMeas
+
+        mps = [
+            np.array([[0.0, 0.107], [0.994, 0.0]]),
+            np.array(
+                [
+                    [[0.0, 0.0, 0.0, -0.0], [1.0, 0.0, 0.0, -0.0]],
+                    [[0.0, 1.0, 0.0, -0.0], [0.0, 0.0, 0.0, -0.0]],
+                ]
+            ),
+            np.array(
+                [
+                    [[-1.0, 0.0], [0.0, 0.0]],
+                    [[0.0, 0.0], [0.0, 1.0]],
+                    [[0.0, -1.0], [0.0, 0.0]],
+                    [[0.0, 0.0], [1.0, 0.0]],
+                ]
+            ),
+            np.array([[-1.0, -0.0], [-0.0, -1.0]]),
+        ]
+        num_wires = 4
+
+        def circuit():
+            _mps_prep_decomposition(
+                *mps, wires=range(2, num_wires + 2), work_wires=[0, 1], right_canonicalize=True
+            )
+
+        plxpr = qml.capture.make_plxpr(circuit)()
```

Comment:
```suggestion
        def circuit(*_mps):
            _mps_prep_decomposition(
                *_mps, wires=range(2, num_wires + 2), work_wires=[0, 1], right_canonicalize=True
            )

        plxpr = qml.capture.make_plxpr(circuit)(*mps)
```
This way `mps` is actually a traced variable.

Reply:
aha ok